src/object_details.rs

```diff
@@ -0,0 +1,48 @@
+use glib;
+use std::fmt::Display;
+use std::fmt::Formatter;
+use std::fmt::Error;
+
+/// Details of an object in an OSTree repo. It contains information about if
+/// the object is "loose", and contains a list of pack file checksums in which
+/// this object appears.
+#[derive(Debug)]
+pub struct ObjectDetails {
+    loose: bool,
+    object_appearances: Vec<String>,
+}
+
+impl ObjectDetails {
+    /// Create a new `ObjectDetails` from a serialized representation.
+    pub fn new_from_variant(variant: glib::Variant) -> Option<ObjectDetails> {
+        let deserialize = variant.get::<(bool, Vec<String>)>()?;
+        Some(ObjectDetails {
+            loose: deserialize.0,
+            object_appearances: deserialize.1,
+        })
+    }
+
+    /// is object available "loose"
+    pub fn is_loose(&self) -> bool {
+        self.loose
+    }
+
+    /// Provide list of pack file checksums in which the object appears
+    pub fn appearances(&self) -> &Vec<String> {
+        &self.object_appearances
+    }
+
+    /// Format this `ObjectDetails` as a string.
+    fn to_string(&self) -> String {
+        format!("Object is {} loose and appears in {} checksums",
+                if self.loose {"available"} else {"not available"},
```

Comment:
This looks somehow wrong. Were you maybe thinking of doing `"not"` and `""`?

Reply:
I did this on purpose, because I do not like space handling of this `""` / `"not "` solutions:
` format!("Object is {}loose and appears in {} checksums", if self.loose {""} else {"not "},`
If this is the preferred solution i will change it that way, but I'm quite unsure what to print anyway. Would it be ok to have just something like this:
` format!("{:?}", self)`

Reply:
No preferences, but I do have some problem parsing your "Object is not available loose...".
I'd be ok with "loose" / "not loose".

Reply:
> [...] I do have some problem parsing your "Object is not available loose..." [..]
Same her while parsing https://ostreedev.github.io/ostree/reference/ostree-OstreeRepo.html#OSTREE-REPO-LIST-OBJECTS-VARIANT-TYPE:CAPS from where the wording is borrowed.
I'm ok with "loose" and "not loose" either.